zulipterminal/ui_tools/boxes.py

```diff
@@ -1637,37 +1637,63 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
             self.model.controller.view.write_box.msg_write_box.set_edit_pos(len(quote))
             self.model.controller.view.middle_column.set_focus("footer")
         elif is_command_key("EDIT_MESSAGE", key):
-            if self.message["sender_id"] != self.model.user_id:
-                self.model.controller.report_error(
-                    " You can't edit messages sent by other users."
-                )
+            if (
+                self.message["sender_id"] != self.model.user_id
+                and self.message["subject"] != "(no topic)"
+            ):
+                if self.message["type"] == "stream":
+                    self.model.controller.report_error(
+                        " You can't edit messages sent by other users"
+                        " with non-empty topic."
+                    )
```

Comment:
This should be the accurate definition of "(no topic)" messages i.e. empty topic. But, I am a bit hesitant to print this in the footer (the user might not know how we represent empty topics?).
I also had `You can't edit messages sent by other users which do not have "(no topic)"`

Reply:
After a short discussion with @prah23 via PM's we've decided to go with "You can't edit messages sent by other users that already have a topic." :)